src/mlpack/methods/ann/layer/replicate.hpp

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file methods/ann/layer/replicate.hpp
+ * @author Adam Kropp
+ *
+ * Definition of the Replicate class, which replicates the input n times
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_REPLICATE_HPP
+#define MLPACK_METHODS_ANN_LAYER_REPLICATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include "layer.hpp"
+
+namespace mlpack {
+
+/**
+ * Implementation of the Replicate class. The Replicate class replicates the
+ * input a specified number of times along each axis.  The output will have the
+ * same number of dimnensions as the input, with each dimension multiplied by
+ * the specified scalar.
+ *
+ * @tparam MatType Matrix representation to accept as input and use for
+ *    computation.
+ */
+template <typename MatType = arma::mat>
+class ReplicateType : public Layer<MatType>
+{
+ public:
+  /**
+   * Create the Replicate object with no multiples.  If not set later, this
+   * is equivalent to an identity layer.
+   */
+  ReplicateType();
+
+  /**
+   * Create the Replicate object, specifying the number of repeated blocks
+   * along each dimension.
+   *
+   * @param multiples The number of times to repeat along each axis. Must be
+   *        the same size as InputDimensions.
+   */
+  ReplicateType(std::vector<size_t> multiples);
+
+  /**
+   * Destroy the layers held by the model.
+   */
+  virtual ~ReplicateType();
```

Comment:
Feel free to save a little space in the `_impl.hpp` file by just adding `{ }` here, but, up to you. :+1:

Reply:
👍